src/hip_memory.cpp

```diff
@@ -1266,8 +1266,12 @@ hipError_t hipMemcpyToSymbol(void* dst, const void* src, size_t count,
 
     tprintf(DB_MEM, " symbol '%s' resolved to address:%p\n", symbol_name, dst);
 
-    return ihipLogStatus(
-        hipMemcpy(static_cast<char*>(dst) + offset, src, count, kind));
+    if (dst == nullptr) {
+        return ihipLogStatus(hipErrorInvalidSymbol);
+    }
+
+    return ihipLogStatus(hip_internal::memcpySync((char*)dst+offset, src, count, kind,
```

Comment:
I would prefer to have cpp style casting e.g. static_cast<char*> over (char*)

Reply:
Agreed.